frontend_v2/src/app/components/publiclists/challengelist/challengelist.component.ts

```diff
@@ -226,15 +241,24 @@ export class ChallengelistComponent implements OnInit {
     this.routerPublic = this.router;
   }
 
+  /**
+   * After view is initialized.
+   */
+   ngAfterViewInit() {}
+
+ 
   /**
    * Listener for page scroll event
    */
   @HostListener('window:scroll', [])
   onWindowScroll(): void {
+    setTimeout(() => {
```

Comment:
We won't require to use `setTimeout` anywhere as the error was coming only because the variable `ongoingChallenges` was getting used in the template even before its initialising.